### ISSUE TITLE
shadow_unrelated triggers on underscore name binding

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#3624]
````rust
#![warn(clippy::shadow_unrelated)]
fn main() {
    let _a = 3;
    let _a = 4;
}
````

````
warning: `_a` is shadowed by `4`
 --> src/main.rs:5:9
  |
5 |     let _a = 4;
  |         ^^
  |
note: lint level defined here
 --> src/main.rs:2:9
  |
2 | #![warn(clippy::shadow_unrelated)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^
note: initialization happens here
 --> src/main.rs:5:14
  |
5 |     let _a = 4;
  |              ^
note: previous binding is here
 --> src/main.rs:4:9
  |
4 |     let _a = 3;
  |         ^^
````

maybe it should not warn for anything that starts with an underscore?

**Comment by detrumi** [comment]
I don't think it should, because it's still shadowing, and might be used even if it starts with an underscore.

It doesn't trigger for `_` itself, at least.

**Comment by matthiaskrgr** [comment]
Right, there was a thinko at my side, sorry about that.
